src/game/state/player-state.js

```diff
@@ -237,4 +238,10 @@ function noteBuffer (state) {
   }
 }
 
+function isBeginnerChart (notechart) {
+  const info = notechart.songInfo
+  const insane = info.difficulty >= 5
+  return !insane && info.level <= 2
```

Comment:
Some charts have level 0 (but with normal difficulty) for insane. Some charts even have negative level (!). Scope this to level 1, 2, 3, 4. Add special case for tutorial song…